common_python/constants.py

```diff
@@ -200,56 +200,27 @@
-#     natural=coastline, nosea, sea, beach, land, scrub, water, wetland =wood \
-#     leisure=park, nature_reserve \
-#     railway=abandoned, bus_guideway, disused, funicular, light_rail, miniature, narrow_gauge, preserved, rail, subway, tram \
-#     surface \
-#     tracktype \
-#     tunnel \
-#     waterway=canal, drain, river, riverbank \
-#     wood=deciduous'
-
-FILTERED_TAGS_NAMES = ['admin_level=2', 'area=yes', 'mountain_pass', 'natural',
-                       'place=city, hamlet, island, isolated_dwelling, islet, locality, suburb, town, village, country']
-
-# FILTERED_TAGS_NAMES = 'admin_level=2 \
-#     area=yes \
-#     mountain_pass \
-#     natural \
-#     place=city, hamlet, island, isolated_dwelling, islet, locality, suburb, town, village, country'
-
-FILTERED_TAGS_WIN = 'access= \
-    area=yes \
-    bicycle= \
-    bridge= \
-    foot=ft_yes =foot_designated \
-    highway=abandoned =bus_guideway =disused =bridleway =byway =construction =cycleway =footway =living_street =motorway =motorway_link =path =pedestrian =primary =primary_link =residential =road =secondary =secondary_link =service =steps =tertiary =tertiary_link =track =trunk =trunk_link =unclassified \
-    natural=coastline =nosea =sea =beach =land =scrub =water =wetland =wood \
-    leisure=park =nature_reserve \
-    railway=abandoned =bus_guideway =disused =funicular =light_rail =miniature =narrow_gauge =preserved =rail =subway =tram \
-    surface= \
-    tracktype= \
-    tunnel= \
-    waterway=canal =drain =river =riverbank \
-    wood=deciduous'
-
-FILTERED_TAGS_WIN_NAMES = 'admin_level=2 \
-    area=yes \
-    mountain_pass= \
-    natural= \
-    place=city =hamlet =island =isolated_dwelling =islet =locality =suburb =town =village =country'
+TAGS_TO_KEEP_UNIVERSAL = {
+    'access': '',
+    'area': 'yes',
+    'bicycle': '',
+    'bridge': '',
+    'foot': ['ft_yes', 'foot_designated'],
+    'highway': ['abandoned', 'bus_guideway', 'disused', 'bridleway', 'byway', 'construction', 'cycleway', 'footway', 'living_street', 'motorway', 'motorway_link', 'path', 'pedestrian', 'primary', 'primary_link', 'residential', 'road', 'secondary', 'secondary_link', 'service', 'steps', 'tertiary', 'tertiary_link', 'track', 'trunk', 'trunk_link', 'unclassified'],
+    'natural': ['coastline', 'nosea', 'sea', 'beach', 'land', 'scrub', 'water', 'wetland', 'wood'],
+    'leisure': ['park', 'nature_reserve'],
+    'railway': ['abandoned', 'bus_guideway', 'disused', 'funicular', 'light_rail', 'miniature', 'narrow_gauge', 'preserved', 'rail', 'subway', 'tram'],
+    'surface': '',
+    'tracktype': '',
+    'tunnel': '',
+    'waterway': ['canal', 'drain', 'river', 'riverbank'],
+    'wood': 'deciduous'
+}
+
+NAME_TAGS_TO_KEEP_UNIVERSAL = {
```

Comment:
Can you (or anybody else) remember why we need two distinct sets resulting in two filtered files which are then merged with land and sea on the tile processing/generation?

Was it because of CLI argument length limitations on some platform?

Reply:
I looked it up.

We introduced the _NAMES constants and corresponding osmium / osmfilter processing in #46.
There are some sentenced about filtering out names of highways etc. with these additions.

I think lately there was a discussion about the need of these two sets in Telegram with no real outcome. In my opinion there might not be a technical need for two sets.

I'll try if my unittests check if there is a knowingly error. If yes, I could refactor that part in another PR.

Reply:
Ah, thanks. I think I have not fully understood the filtering behaviour of the tools entirely.

At least on the non-Windows side this has no effect as the second `osmium tags-filter` operation does not explicitely list a `name=` filter expression like the second `osmfilter` operation does on the Windows side.
But adding it, the filtered germany map for names blows up from 263MB to 1109MB.

No matter what, names are present when checking with `osmium tags-count` or a custom `vmt-elemnt.xml` theme displaying highway names. So I dunno if the two tag-sets are really needed.

Perhaps we can take this discussion to a new issue later on.

Reply:
Just as a heads up ... I just did a little testing. Using osmfilter to only keep tag `historic`, making sure it's rendered by osmiums mapwriter plugin and the name tag of the `historic` node I am looking at is displayed just fine. So I dunno if this is just `osmfilter` magic with that special name handling, on the osmium one wouldn't need two sets of tags. (I might still be completely missing something here.)

Reply:
we'll discuss that in #109